### PR TITLE
integration of connext amarok

### PR DESCRIPTION
**Pull request by marie-fourier**
Updated hardhat to 2.9.8, because forking was not working in the old version (Ref: https://github.com/NomicFoundation/hardhat/releases/tag/2.9.2)

Changed diamond contracts solidity version to dynamic ">=0.8.4 <0.9.0". LibDiamond.sol will be used by many facets that may have different versions of solidity because they depend on external packages. For example, AmarokFacet makes use of nxtp/contracts package which is written in 0.8.11. Other bridges will use another package written in different version of solidity

* Tests must be performed in goerli chain hardfork

### CODE REVIEW

**Comment by zincoshine** [comment]
@marie-fourier is this ready for review?

File: package.json [hunk 1]
```diff
@@ -68,9 +68,11 @@
     "test": "hardhat test"
   },
   "devDependencies": {
+    "@connext/nxtp-contracts": "^0.2.0-beta.10",
```
Comment (zincoshine) [inline package.json:71]: Can we pls use an exact version of an external package?

File: package.json [hunk 2]
```diff
@@ -68,9 +68,11 @@
...
     "@nomiclabs/hardhat-etherscan": "2.1.7",
     "@nomiclabs/hardhat-web3": "2.0.0",
     "@openzeppelin/contracts": "4.6.0",
+    "@openzeppelin/contracts-upgradeable": "^4.6.0",
```
Comment (zincoshine) [inline package.json:75]: Same as above an exact version. These external packages are not backward compatible and if a new version is released our build will break.

File: src/bridges/facets/AmarokFacet.sol [hunk 3]
```diff
@@ -0,0 +1,328 @@
...
+/**
+ * @title Connext Amarok Integration
+ *
+ * @notice Contract which provides bridging functionality through Connext
+ *
+ */
+contract AmarokFacet {
+  // storage
+  
+  bytes32 internal constant NAMESPACE = hex"a4cabcf91d6212ba2873e2f19de6e26ae65fd3280b9eaa6d8bc38c5fba52ec9d"; //keccak256("io.etherspot.facets.amarok");
```
Comment (zincoshine) [inline src/bridges/facets/AmarokFacet.sol:18]: Is this keccak256("io.etherspot.facets.amarok");

Comment (zincoshine) [reply src/bridges/facets/AmarokFacet.sol:18]: Can't we use etherspot-specific namespace?

File: src/bridges/facets/AmarokFacet.sol [hunk 4]
```diff
@@ -0,0 +1,328 @@
...
+  struct Storage {
+    address connext;
+    uint32 chainId;
```
Comment (zincoshine) [inline src/bridges/facets/AmarokFacet.sol:21]: From solidity 0.8.0 we can get the current chainId using `block.chainid`. Do we then need this variable in storage?

File: src/bridges/facets/AmarokFacet.sol [hunk 5]
```diff
@@ -0,0 +1,328 @@
...
+import { IERC20 } from "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import { LibDiamond } from "../libs/LibDiamond.sol";
+
+/**
+ * @title Connext Amarok Integration
+ *
+ * @notice Contract which provides bridging functionality through Connext
+ *
+ */
+contract AmarokFacet {
```
Comment (zincoshine) [inline src/bridges/facets/AmarokFacet.sol:15]: Amarok is just a version name of the Connext product. I suggest we rename the contract to ConnextFacet or NxtpFacet

File: src/bridges/facets/AmarokFacet.sol [hunk 6]
```diff
@@ -0,0 +1,328 @@
...
+    uint32 chainId;
+  }
+
+  // events
+
+  /**
+   * @dev Emitted when facet initializes
+   * @param _connext connext handler address
+   */
+  event AmarokInitialized(address _connext);
```
Comment (zincoshine) [inline src/bridges/facets/AmarokFacet.sol:30]: Pls see above on the contract name and change events and function names accordingly.

**Review by zincoshine — CHANGES_REQUESTED** [review]
Does Connext contract provide the list of tokens supported by them on a given chain for bridging?

**Comment by zincoshine** [comment]
Can't you get tests working on Etherspot chain? We can configure Etherspot chain with the necessary changes if required.

File: src/bridges/facets/AmarokFacet.sol (hunk 3 above)
Comment (lukewickens1989) [reply src/bridges/facets/AmarokFacet.sol:18]: Do we need this? As there is no diamond storage being used can we just store this as variables inside the contract?

File: src/bridges/facets/AmarokFacet.sol (hunk 4 above)
Comment (lukewickens1989) [reply src/bridges/facets/AmarokFacet.sol:21]: Agreed.

File: src/bridges/facets/AmarokFacet.sol (hunk 3 above)
Comment (zincoshine) [reply src/bridges/facets/AmarokFacet.sol:18]: I thought the hex was used to pull up a specific version of the contract code from storage?

File: src/bridges/facets/AmarokFacet.sol (hunk 4 above)
Comment (marie-fourier) [reply src/bridges/facets/AmarokFacet.sol]: yes, it actually should be named `domainId` and it is different from chain id

File: src/bridges/facets/AmarokFacet.sol (hunk 3 above)
Comment (marie-fourier) [reply src/bridges/facets/AmarokFacet.sol]: >Is this keccak256("io.etherspot.facets.amarok");

yes

**Comment by marie-fourier** [comment]
> Can't you get tests working on Etherspot chain? We can configure Etherspot chain with the necessary changes if required.

@zincoshine Integration contracts must interact with connext's contracts in goerli chain. If we can hardfork goerli in our etherspot chain, then we can use it

File: package.json (hunk 2 above)
Comment (marie-fourier) [reply package.json]: fixed

File: src/bridges/facets/AmarokFacet.sol (hunk 3 above)
Comment (lukewickens1989) [reply src/bridges/facets/AmarokFacet.sol]: Sorry yes you are right

**Comment by zincoshine** [comment]
deploy/203_deploy_amarok_facet.ts should be renamed to deploy/203_deploy_connext_facet.ts

File: extensions/constants.ts [hunk 7]
```diff
@@ -63,7 +63,7 @@ export const NETWORK_CONFIGS: {
   },
   [NetworkNames.Goerli]: {
     chainId: 5,
-    defaultProviderUrl: 'infura',
+    defaultProviderUrl: 'https://goerli.infura.io/v3/000a856f3b024bcba2f5c507fe62738d',
```
Comment (zincoshine) [inline extensions/constants.ts:66]: Do not store Infura keys in the repo. Always use environment variables for these things.

**Review by zincoshine — CHANGES_REQUESTED** [review]

File: extensions/constants.ts (hunk 7 above)
Comment (marie-fourier) [reply extensions/constants.ts]: removed

**Review by zincoshine — CHANGES_REQUESTED** [review]
@marie-fourier `src/bridges/facets/AmarokFacet.sol` file name has to changed to `ConnextFacet.sol`

**Comment by zincoshine** [comment]
Please check all such references to Amarok and change it to Connext

**Review by lukewickens1989 — APPROVED** [review]

**Review by riccardopillar — APPROVED** [review]

**Pull request merged**